scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+                getOut().println("path: " + path.toString());
+                if (entry.isDirectory()) {
+                    Files.createDirectories(path);
+                } else {
+                    Path parentDir = path.getParent();
+                    getOut().println("parentDir of path: " + parentDir.toString());
+                    if (!Files.exists(parentDir)) {
+                        Files.createDirectories(parentDir);
+                    }
+                    if (Files.exists(path)) {
+                        new File(path.toString()).delete();
+                    }
+
+                    Files.copy(zipFile.getInputStream(entry), path);
+                }
+            }
+        }
+    }
+
+    private void FtpPut(FTPClient ftpClient) throws IOException, IllegalArgumentException {
+        if (ftpLocalFilePath.isEmpty()) {
+            throw new IllegalArgumentException("The ftp local file path variable is empty. Please, enter a value.");
+        }
+        getOut().println("Exporting file(s) from " + ftpLocalFilePath + " to " + ftpRemoteFilePath);
+        File localFile = new File(ftpLocalFilePath);
+        // if the ftp remote file path does not exist then it is created.
+        if (!checkDirectoryExists(ftpClient, ftpRemoteFilePath)) {
+            ftpClient.makeDirectory(ftpRemoteFilePath);
+            getOut().println(ftpRemoteFilePath + " does not exist. Therefore, it is created.");
+        } else {
+            getOut().println(ftpRemoteFilePath + " exists.");
+        }
+        // If it is a single file:
+        if (localFile.isFile()) {
+            getOut().println("A single FILE to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, ftpLocalFilePath, localFile.getName());
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+            } else {
+                getOut().println("COULD NOT upload the file: " + ftpLocalFilePath);
+            }
+        }
+        // If it is a folder, upload all its contents recursively
+        else if (localFile.isDirectory()) {
+            getOut().println("A DIRECTORY to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            uploadDirectory(ftpClient, "", ftpLocalFilePath, "");
+        } else {
+            throw new IllegalArgumentException(localFile + " not found. Please, enter a valid path.");
+        }
+        getOut().println("END Export file(s) to FTP.");
+    }
+
+    private void uploadDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
+            String remoteParentDir) throws IOException {
+
+        getOut().println("LISTING directory: " + localParentDir);
+
+        File localDir = new File(localParentDir);
+        File[] subFiles = localDir.listFiles();
+        if (subFiles != null && subFiles.length > 0) {
```

Comment:
I think you no need this if at all. For each loop does everything for you

Reply:
done